src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+
+import { Person } from '../../types/Person';
+import { peopleFromServer } from '../../data/people';
+import classNames from 'classnames';
+
+type Props = {
+  onSelect: (person: Person | null) => void;
+  delay: number;
+};
+
+export const Dropdown: React.FC<Props> = ({ onSelect, delay }) => {
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), []);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+    setIsDropdownActive(true);
+    onSelect(null);
+  };
+
+  const handleClick = (person: Person) => {
+    onSelect(person);
+    setQuery(person.name);
+    setIsDropdownActive(false);
+  };
+
+  const filterPeople = useMemo(() => {
```

Comment:
```suggestion
  const filteredPeople = useMemo(() => {
```
it's not a function, but already filtered array of people